Importing libraries and Dataset

In [5]:
import pandas as pd
import numpy as np

In [6]:
ratings = pd.read_csv("/content/teachers_rating_dataset.csv")
ratings.head()

Prof  Gender Tenure  Beauty  Rating  Students  Age        Division
0  Prof_52  Female     No    5.66    4.11        62   26  Upper Division
1  Prof_93    Male    Yes    7.28    3.86        95   37  Lower Division
2  Prof_15  Female     No    7.02    4.22        84   35  Upper Division
3  Prof_72    Male    Yes    8.15    4.57       300   47  Lower Division
4  Prof_61  Female     No    6.68    4.25       131   40  Lower Division

Q1) Using the teachers' rating dataset, what is the probability of receiving an evaluation score of greater than 4.5

In [7]:
prob_greater_4_5 = (ratings["Rating"] > 4.5).mean()
print(f"Probability of receiving an evaluation score greater than 4.5: {prob_greater_4_5:.3f}")

Probability of receiving an evaluation score greater than 4.5: 0.167


-> There is approximately a 16.7% probability that a professor receives a rating greater than 4.5.

Q2. Using the teachers' rating dataset, what is the probability of receiving an evaluation score greater than 3.5 and less than 4.2

In [8]:
prob_between = ((ratings["Rating"] > 3.5) & (ratings["Rating"] < 4.2)).mean()
print(f"Probability of receiving an evaluation score between 3.5 and 4.2: {prob_between:.3f}")

Probability of receiving an evaluation score between 3.5 and 4.2: 0.517


-> There is approximately a 51.7% probability that a professor’s evaluation score falls between 3.5 and 4.2.

Q3. Using the two-tailed test from a normal distribution:
1. A professional basketball team wants to compare its performance with that of players in a regional league.
2. The pros are known to have a historic mean of 12 points per game with a standard deviation of 5.5.
3. A group of 36 regional players recorded on average 10.7 points per game.
4. The pro coach would like to know whether his professional team scores on average are different from that of the regional players.
State the null hypothesis
Null Hypo: ("The mean point of the regional players is not different from the historic mean")
Alternative Hypo: ("The mean point of the regional players is different from the historic mean")

Null Hypothesis (H₀): μ = 12
-> “The mean point of the regional players is not different from the historic mean.”

Alternative Hypothesis (H₁): μ ≠ 12
-> “The mean point of the regional players is different from the historic mean.”

This is a two-tailed test.

Computing the z-score Test:

In [9]:
import math
import scipy.stats as stats
mu = 12
sigma = 5.5
x_bar = 10.7
n = 36
z = (x_bar - mu) / (sigma / np.sqrt(n))
print(f"Z-score: {z:.3f}")
p_value = 2 * (1 - stats.norm.cdf(abs(z)))
print("Two-tailed p-value:", p_value)
alpha = 0.05
if p_value < alpha:
    print("Reject the null hypothesis: There is a significant difference.")
else:
    print("Fail to reject the null hypothesis: No significant difference.")

Z-score: -1.418
Two-tailed p-value: 0.15613768876786005
Fail to reject the null hypothesis: No significant difference.


The two-tailed p-value corresponding to z = -1.42 is approximately 0.156.
Since p-value (0.156) > 0.05


We fail to reject the null hypothesis.

There is no statistically significant difference between the average points scored by the regional players (10.7) and the historic mean of 12 points per game.
In other words, the regional players’ scoring performance is not significantly different from that of the professional team.